### PR TITLE
kickoff infrastructure, CI, & packaging

### LINKED ISSUE #3: update infrastructure
***update infrastructure***

Ad top-level issues here & tag back in individual tickets

* [ ] "packagize" here as over in `simplification`
* [ ] full CI matrix - OS & min/latest/dev
* [ ] etc...

### LINKED ISSUE #8: testing matrix for Python 3.10 & above
Currently [`Python 3.11`](https://github.com/uscuni/simplification/blob/e3411a4a376b217a40c565559553e381914d42e1/pyproject.toml#L28) is pinned over in [`simplification`](https://github.com/uscuni/simplification) due to [`graphblas`](https://github.com/uscuni/simplification/issues/32).  This is not a problem [here](https://github.com/uscuni/sgeop/blob/dc728e10fbf17d365b90d0c5031c8287596b46d6/pyproject.toml#L27). 

XREF #3

### PR DESCRIPTION
**Pull request by jGaboardi**
* kickoff infrastructure, CI, & packaging
* resolves #3 
* resolves #8 
* ignore Windows set up failure for now

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+         uses: mamba-org/setup-micromamba@v2
+         with:
+           environment-file: ${{ matrix.environment-file }}
+           micromamba-version: "latest"
+
+       # - must be editable install for current tests to pass
+       # - due to data & pathing; probably need to:
+       #    1. rethink current 'convenience' funcs;
+       #    2. move `data/` into `core/data/`
+       #    3. something else
```
Comment (martinfleis) [inline .github/workflows/testing.yml:61]: Is this still relevant?

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+       - name: install package
+         run: "pip install -e .[tests]"
```
Comment (martinfleis) [inline .github/workflows/testing.yml:63]:
```suggestion
         run: "pip install -e . --no-deps"
```

we should not install test deps with pip

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+       - name: environment info
+         run: "micromamba info && micromamba list"
+
```
Comment (martinfleis) [inline .github/workflows/testing.yml:67]:
```suggestion
```
This is done automatically within setup-micromamba step

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+files: "sgeop\/|notebooks\/"
```
Comment (martinfleis) [inline .pre-commit-config.yaml:1]:
```suggestion
files: "sgeop\/"
```

I don't expect notebooks folder to be in this repo. We may have notebooks in docs but let's add those once docs land.

File: README.md [hunk 5]
```diff
@@ -1,9 +1,31 @@
...
+
+## Examples
+
+## Contribution
+
+## Recommended Citation
+
+---------------------------------------
+
+Copyright (c) 2023-, SGeoP Developers
```
Comment (martinfleis) [inline README.md:31]: Is SGeoP capitalisation we want? I'd maybe stick to sgeop but fine with either. cc @anastassiavybornova

Comment (martinfleis) [inline README.md:31]:
```suggestion
Copyright (c) 2024-, SGeoP Developers
```

File: ci/py310_sgeop-oldest.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  - pyogrio
+  - scipy
+  - shapely
+  # testing
+  - pytest
+  - pytest-cov
+  - pytest-xdist
+  - pip:
+    # from bleeding edge functionality
+    - git+https://github.com/pysal/momepy.git
```
Comment (martinfleis) [inline ci/py310_sgeop-oldest.yaml:26]: That is due to https://github.com/pysal/momepy/pull/652 right? I will cut a patch 0.8.1 to pin that.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (jGaboardi) [reply .github/workflows/testing.yml:61]: Not exactly currently, but may be in the coming time. Or maybe not? This goes back to data reading stuff. Let's remove now and I can put back in if needed.

File: .github/workflows/testing.yml (hunk 2 above)
Comment (jGaboardi) [reply .github/workflows/testing.yml:63]: Good point for here. That was needed for the original implementation.

File: README.md (hunk 5 above)
Comment (anastassiavybornova) [reply README.md]: lowercase please!! ❤️

**Review by anastassiavybornova — APPROVED** [review]

File: ci/py310_sgeop-oldest.yaml (hunk 6 above)
Comment (martinfleis) [reply ci/py310_sgeop-oldest.yaml:27]: https://github.com/pysal/momepy/releases/tag/v0.8.1

**Comment by jGaboardi** [comment]
waiting for `momepy=0.8.1` to show up on `conda-forge`

**Comment by jGaboardi** [comment]
[`v0.8.1`](https://github.com/conda-forge/momepy-feedstock/pull/39) just cut from the `feedstock` side. Let's re-run CI once it shows up on `conda-forge`.

**Review by martinfleis — APPROVED** [review]

**Review by martinfleis — APPROVED** [review]

**Pull request merged**